job/dependency_resolver.go

```diff
@@ -52,6 +55,131 @@ func (r *dependencyResolver) Resolve(ctx context.Context, projectSpec models.Pro
 	return jobSpec, nil
 }
 
+// ResolveAndPersist resolve inter/intra dependencies inferred by optimus and persist
+func (r *dependencyResolver) ResolveAndPersist(ctx context.Context, projectSpec models.ProjectSpec, jobSpec models.JobSpec,
+	observer progress.Observer) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	projectJobSpecRepo := r.projectJobSpecRepoFactory.New(projectSpec)
+
+	// resolve inter/intra dependencies inferred by optimus
+	jobSpec, err := r.resolveInferredDependencies(ctx, jobSpec, projectSpec, projectJobSpecRepo, observer)
+	if err != nil {
+		return err
+	}
+
+	// resolve statically defined dependencies
+	jobSpec, err = r.resolveStaticDependencies(ctx, jobSpec, projectSpec, projectJobSpecRepo)
+	if err != nil {
+		return err
+	}
+
+	return r.persistDependencies(ctx, projectSpec, jobSpec)
+}
+
+func (r *dependencyResolver) persistDependencies(ctx context.Context, projectSpec models.ProjectSpec, jobSpec models.JobSpec) error {
+	// delete from dependency table
+	if err := r.dependencyRepo.DeleteByJobID(ctx, jobSpec.ID); err != nil {
+		return err
+	}
+
+	for _, dependency := range jobSpec.Dependencies {
+		// insert the new ones
+		err := r.dependencyRepo.Save(ctx, projectSpec.ID, jobSpec.ID, dependency)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *dependencyResolver) FetchJobDependencies(ctx context.Context, projectSpec models.ProjectSpec,
```

Comment:
FetchJobDependencies I believe can be simplified the logic of fetching and updating with inter/intra job dependencies .

Reply:
have modified this, FetchJobDependencies will just fetch the dependencies, adapting to the required model is handled on the enriching part (in deployer).